src/pages/index.js

```diff
@@ -23,45 +34,182 @@ import {
+
+    // load user bio from server
+    const name = serverUserInfo.name;
+    const job = serverUserInfo.about;
+    const avatar = serverUserInfo.avatar;
+    // add user bio to DOM
+    userProfile.setUserInfo(name, job);
+    userProfile.setAvatar(avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 // handler
+
 function handleAssignEditInput() {
   const userProfileData = userProfile.getUserInfo();
   editPopup.setInputValues(userProfileData);
 }
 
+function handleAvatarSubmit(inputObj) {
+  avatarEditPopup.showUploadingState();
+  const avatarLink = inputObj[avatarLinkInputId];
+  api
+    .updateProfileImage(avatarLink)
+    .then((res) => {
+      avatarEditPopup.showUploaded();
+      console.log(res);
+      userProfile.setAvatar(res.avatar);
+      avatarEditPopup.close();
+      formValidators[avatarFormId].disableSubmitButton();
+      avatarEditPopup.reset();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      avatarEditPopup.showUploaded();
+    });
+}
+
 function handleProfileSubmit(inputObj) {
-  userProfile.setUserInfo(
-    inputObj[editInputTitleId],
-    inputObj[editInputDescriptionId]
-  );
-  editPopup.close();
+  const nameEl = inputObj[editInputTitleId];
+  const jobEl = inputObj[editInputDescriptionId];
+  editPopup.showUploadingState();
+  api
+    .updateUserInfo({
+      editFormNameInput: nameEl,
+      editFormAboutInput: jobEl,
+    })
+    .then((res) => {
+      editPopup.showUploaded();
+      userProfile.setUserInfo(res.name, res.about); // get userinfo from server and apply to profile
+      editPopup.close();
+      formValidators[profileEditFormId].disableSubmitButton();
+      editPopup.reset();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
 }
 function creatCard(item) {
-  const cardElement = new Card(item, cardTempSelector, handleImageClick);
+  const cardElement = new Card(
+    item,
+    cardTempSelector,
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick
+  );
   const cardItem = cardElement.getView();
   return cardItem;
 }
+
 function handleAddCardSubmit(inputObj) {
-  addCardPopup.close();
-  const newCard = creatCard({
-    name: inputObj[addCardInputTitleId],
-    link: inputObj[addCardInputLinkId],
-  });
-  cardList.addItem(newCard);
-  formValidators[addCardFormId].disableSubmitButton();
-  addCardPopup.reset();
+  const cardName = inputObj[addCardInputTitleId];
+  const cardLink = inputObj[addCardInputLinkId];
+  addCardPopup.showUploadingState();
+  api
+    .addNewCard({ cardElementName: cardName, cardElementLink: cardLink })
+    .then((card) => {
+      addCardPopup.showUploaded();
+      const newCard = creatCard(card);
+      cardList.addItem(newCard);
+      formValidators[addCardFormId].disableSubmitButton();
+      addCardPopup.close();
+      addCardPopup.reset();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      addCardPopup.showToUpload;
```

Comment:
Needs correction: don't forget to call the function

Also use ` addCardPopup.showUploaded();` here as you have no `showToUpload` function